includes/classes/Indexable/Post/Post.php

```diff
@@ -95,14 +97,65 @@ public function query_db( $args ) {
 		 */
 		$args = apply_filters( 'ep_index_posts_args', $args );
 
+		// Implement custom pagination for better performance.
+		if ( ! empty( $args['ep_indexing_advanced_pagination'] ) ) {
+			// Apply an initial start_post_id if needed.
+			$args['ep_indexing_start_post_id'] = ! empty( $args['ep_indexing_start_post_id'] ) ? $args['ep_indexing_start_post_id'] : PHP_INT_MAX;
```

Comment:
Discussed in slack, but just noting here for historical reasons that WP is pretty strict about posts not going over PHP_INT_MAX. https://github.com/WordPress/WordPress/blob/d0a32c51112bc4f4827476ad417cc8905abcd85c/wp-includes/class-wp-post.php#L234-L237

Going to stick with `PHP_INT_MAX` as the default for now, though I did switch most of the logic to be more supportive in case this is changed in the future.